spec/models/manageiq/providers/vmware/network_manager/refresher_spec.rb

```diff
@@ -30,205 +63,352 @@
+              :cidr   => '10.12.0.24/16',
+              :shared => false,
+              :type   => network_type_vdc
+            )
+            expect(vdc_net.cloud_subnets.count).to eq(1)
+          end
 
-    def assert_specific_subnet
-      expect(vapp_subnet).to have_attributes(
-        :name                  => 'subnet-vApp network test (RHEL7-web-002)',
-        :cidr                  => '192.168.2.1/24',
-        :dhcp_enabled          => true,
-        :gateway               => '192.168.2.1',
-        :dns_nameservers       => [],
-        :type                  => subnet_type,
-        :ext_management_system => @ems_network,
-        :cloud_network         => vapp_net,
-        :network_router        => router
-      )
-      expect(vapp_subnet.network_ports.count).to eq(1)
-      expect(vapp_subnet.vms.count).to eq(1)
-    end
+          def assert_specific_subnet
+            expect(vdc_subnet).to have_attributes(
+              :name                           => 'subnet-ManageIQ Dev external network',
+              :cidr                           => '10.12.0.24/16',
+              :status                         => nil,
+              :dhcp_enabled                   => nil,
+              :gateway                        => "10.12.0.24",
+              :network_protocol               => nil,
+              :dns_nameservers                => ["10.12.0.13"],
+              :ipv6_router_advertisement_mode => nil,
+              :ipv6_address_mode              => nil,
+              :extra_attributes               => nil,
+              :type                           => subnet_type,
+              :ext_management_system          => @ems_network,
+              :availability_zone              => nil,
+              :cloud_network                  => vdc_net,
+              :cloud_tenant                   => nil,
+              :network_router                 => nil,
+              :network_group                  => nil,
+              :parent_cloud_subnet            => nil
+            )
+            expect(vdc_subnet.network_ports.count).to eq(2)
+            expect(vdc_subnet.vms.count).to eq(2)
+          end
 
-    def assert_specific_network_port
-      expect(net_port).to have_attributes(
-        :name                  => 'vAppRHEL7-w-002#NIC#0',
-        :mac_address           => '00:50:56:01:00:0c',
-        :device_type           => 'VmOrTemplate',
-        :source                => 'refresh',
-        :ext_management_system => @ems_network,
-        :device                => vm
-      )
-    end
+          def assert_specific_network_port
+            expect(net_port).to have_attributes(
+              :name                  => 'vm5#NIC#0',
+              :mac_address           => '00:50:56:01:01:28',
+              :device_type           => 'VmOrTemplate',
+              :ext_management_system => @ems_network,
+              :device                => vm,
+            )
```

Comment:
Please also check that `net_port.cloud_subnets` equals `[vdc_subnet]` from above. Just in order to be sure that this NIC is connected to the correct subnet. Please add such check also in all other network_port checks.